wcfsetup/install/files/lib/acp/form/BoxAddForm.class.php

```diff
@@ -393,6 +393,37 @@ public function readFormParameters()
         if ($this->boxType === 'system') {
             $this->boxController = ObjectTypeCache::getInstance()->getObjectType($this->boxControllerID);
         }
+
+        $this->readConditions();
+    }
+
+    private function readConditions(): void
+    {
+        foreach ($this->groupedConditionObjectTypes as $groupedObjectTypes) {
+            foreach ($groupedObjectTypes as $objectTypes) {
+                if (\is_array($objectTypes)) {
+                    foreach ($objectTypes as $objectType) {
+                        $objectType->getProcessor()->readFormParameters();
+                        if ($objectType->getProcessor() instanceof MultiPageCondition) {
```

Comment:
It should not be possible that this occurs multiple times, but if it does for some reason, then the `pageIDs` will be overriden by an incorrect value, possibly resulting in an hard to find bug.

Can you add a safety check that ensures `pageIDs` will only ever be set once in this loop, throwing an exception if this is not true? You can use `\assert()`.